### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by oliverSCZ**
**I did the following:**

- Created a class Classroom
- Created a class Book
- Created a class Rental
- Created  has-many, belongs-to and many-to-many relationships.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_reader :students
+
+  def initialize(label = '')
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
+  end
```
Comment (er-contreras) [inline classroom.rb]: - [x] When adding a student to a classroom it also sets the classroom for the student:

Something as follow:

```ruby
  def add_student(student)
    @students << student
    student.classroom = self
  end
```

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require './person'
 
 class Student < Person
+  attr_accessor :classroom
+
```
Comment (er-contreras) [inline student.rb:5]: - [x] When setting the classroom for a student it also adds it to the classrooms' students:

- [x] Check for a method that serves as the 'belongs-to' side (a student belongs to a classroom)

Something like:

```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (er-contreras) [inline book.rb]: - [x] The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 4]
```diff
@@ -24,16 +24,13 @@ def correct_name
     name
   end
 
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (er-contreras) [inline person.rb]: - [x] The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @oliverSCZ, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (oliverSCZ) [reply classroom.rb]: Done

File: student.rb (hunk 2 above)
Comment (oliverSCZ) [reply student.rb:5]: Done

File: book.rb (hunk 3 above)
Comment (oliverSCZ) [reply book.rb]: Done

File: person.rb (hunk 4 above)
Comment (oliverSCZ) [reply person.rb]: Done

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @oliverSCZ, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**